core/src/main/java/com/microsoft/applicationinsights/internal/quickpulse/DefaultQuickPulsePingSender.java

```diff
@@ -117,16 +134,16 @@ public QuickPulseStatus ping() {
     }
 
     @VisibleForTesting
-    String getQuickPulsePingUri() {
-        return getQuickPulseEndpoint() + "/ping?ikey=" + getInstrumentationKey();
+    String getQuickPulsePingUri(String endpointURL) {
+        return endpointURL + "/ping?ikey=" + getInstrumentationKey();
```

Comment:
or `endpointPrefix`?
```suggestion
    String getQuickPulsePingUri(String endpoint) {
        return endpoint + "/ping?ikey=" + getInstrumentationKey();
```